### PR TITLE
Refactor header

### PR DESCRIPTION
**Pull request by adamsilver**
## Context

We (@duncanjbrown and I) made this change because the same HTML for the header is used across candidate, provider, support and API doc interfaces.

There was a lot of repeated code, and logic in the view that's better isolated outside of the Header component itself.

This repeated code caused a bug whereby the mobile toggle menu button appeared even when there were no menu items to show/hide.

## Changes proposed in this pull request

We proposed abstracting the header and navigation items into a component that closely resembles the GOV.UK Frontend Header component.

## Link to Trello card

https://trello.com/c/hKy03E02/691-create-header-component

### CODE REVIEW

**Comment by duncanjbrown** [comment]
Added a commit to include yarn in the GitHub Actions build as unit tests touching components, depend upon it, or at least they do now

File: app/components/header_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
+        <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
+          <% navigation_items.each do |nav_item| %>
+            <li class="govuk-header__navigation-item <%= nav_item.active ? 'govuk-header__navigation-item--active' : '' %>">
+              <% if nav_item.href %>
+                <%= link_to nav_item.text, nav_item.href, class: 'govuk-header__link' %>
+              <% else %>
+                <b><%= nav_item.text %></b>
+              <% end %>
+            <% end %>
```
Comment (tvararu) [inline app/components/header_component.html.erb]:
```suggestion
            <% end %>
          </li>
```

File: app/components/header_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      <%= link_to service_name, service_url, class: 'govuk-header__link govuk-header__link--service-name' %>
+      <% if navigation_items.any? %>
+        <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
+        <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
+          <% navigation_items.each do |nav_item| %>
+            <li class="govuk-header__navigation-item <%= nav_item.active ? 'govuk-header__navigation-item--active' : '' %>">
+              <% if nav_item.href %>
+                <%= link_to nav_item.text, nav_item.href, class: 'govuk-header__link' %>
+              <% else %>
+                <b><%= nav_item.text %></b>
```
Comment (tvararu) [inline app/components/header_component.html.erb]:
```suggestion
                <strong><%= nav_item.text %></strong>
```

`b` isn't so widely used anymore.

**Comment by tvararu** [comment]
Have added review comments with minor nits, then fixed them up into the relevant commit.

**Comment by tvararu** [comment]
Took a good look at this on the review app, looks great!

**Pull request merged**